### PR TITLE
chore: Generate global constant and sec scan config for docker images

### PR DESCRIPTION
**Pull request by hisarbalik**
## Description

Any docker image update requires updating several files, such as source code and yaml files. To make the module release and image bumps more straightforward, the docker images are maintained from a single source `.env` file. All other source code and configuration files, like `sec-scanner-config.yaml`, are generated from the `.env` file.

Changes proposed in this pull request (what was done and why):

- Add go code generator to generate docker image constants from `.env` file
- Generate `sec-scanner-config.yaml` from `.env` file
- Update release docs with new steps
- Add go code and `sec-scanner-config.yaml` generation to the `make generate` target
- Load docker images from `.env` file for benchmark tests
- Add code generation to the `check-clean` target to ensure source code and configuration files are up-to-date 

Changes refer to particular issues, PRs or documents:

- 

## Traceability
- [ ] The PR is linked to a GitHub issue.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.
- [x] If the change is user-facing, the documentation has been adjusted.
- [ ] If a CRD is changed, the corresponding Busola ConfigMap has been adjusted.
- [ ] The feature is unit-tested.
- [ ] The feature is e2e-tested.

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -13,3 +13,10 @@ ENV_GARDENER_MAX_NODES=2
 ENV_ISTIO_VERSION=1.11.0
 ENV_K3D_VERSION=v5.4.7
 ENV_GORELEASER_VERSION=v1.23.0
+
+## Default Docker Images
+DEFAULT_FLUENTBIT_EXPORTER_IMAGE="europe-docker.pkg.dev/kyma-project/prod/directory-size-exporter:v20241024-8bc3f6a8"
+DEFAULT_FLUENTBIT_IMAGE="europe-docker.pkg.dev/kyma-project/prod/external/fluent/fluent-bit:3.2.2"
+DEFAULT_OTEL_COLLECTOR_IMAGE="europe-docker.pkg.dev/kyma-project/prod/kyma-otel-collector:0.115.0-main"
+DEFAULT_SELFMONITOR_IMAGE="europe-docker.pkg.dev/kyma-project/prod/tpi/telemetry-self-monitor:3.0.1-857b814"
+DEFAULT_TELEMETRYGEN_IMAGE="ghcr.io/open-telemetry/opentelemetry-collector-contrib/telemetrygen:v0.115.0"
```
Comment (k15r) [inline .env:22]: can we rename the variables for test?
eg
```suggestion
DEFAULT_TEST_TELEMETRYGEN_IMAGE="ghcr.io/open-telemetry/opentelemetry-collector-contrib/telemetrygen:v0.115.0"
```
they would stand out a bit and make them easier to understand

Comment (hisarbalik) [reply .env]: renamed

File: docs/contributor/releasing.md [hunk 2]
```diff
@@ -23,16 +23,13 @@ This release process covers the steps to release new major and minor versions fo
...
-   - `sec-scanners-config.yaml`:  
-     For the images listed in the `protecode` field:
-      - Update the tag of the `telemetry-manager` image with the new module version following the `x.y.z` pattern. For example, `europe-docker.pkg.dev/kyma-project/prod/telemetry-manager:1.0.0`.
-      - Update the tag of the `kyma-otel-collector` image with the new version released from the [opentelemetry-collector-components](https://github.com/kyma-project/opentelemetry-collector-components) repository. For example, `europe-docker.pkg.dev/kyma-project/prod/kyma-otel-collector:0.100.0-1.0.0
-      - Ensure that all other images have the same versions as those used in the `main.go` file.
    - `main.go`:
       - Update the `version` variable with the new module version following the `x.y.z` pattern.
-      - Update the `defaultOTelCollectorImage` variable with the image referenced in `sec-scanners-config.yaml`. For example, `defaultOtelImage = "europe-docker.pkg.dev/kyma-project/prod/kyma-otel-collector:0.104.0-1.20.0-rc1"`
+   - `make generate-env`:
+     Run the `make generate-env` command to update the `sec-scanners-config.yaml`, and source code file with the new image versions.
```
Comment (NHingerl) [inline docs/contributor/releasing.md:32]:
```suggestion
     To update the `sec-scanners-config.yaml` and the source code file with the new image version, run `make generate-env. 
```

File: .golangci.yaml [hunk 3]
```diff
@@ -146,6 +146,8 @@ linters-settings:
         alias: slicesutils
       - pkg: github.com/kyma-project/telemetry-manager/internal/utils/test
         alias: testutils
+      - pkg: github.com/kyma-project/telemetry-manager/internal/constant
```
Comment (skhalash) [inline .golangci.yaml:149]: It is not really needed, right?

File: Makefile [hunk 4]
```diff
@@ -54,6 +54,10 @@ YAMLFMT          := $(TOOLS_BIN_DIR)/yamlfmt
 STRINGER         := $(TOOLS_BIN_DIR)/stringer
 WSL				 := $(TOOLS_BIN_DIR)/wsl
 K3D              := $(TOOLS_BIN_DIR)/k3d
+GENERATE_CONST   := $(TOOLS_BIN_DIR)/generate-const
```
Comment (skhalash) [inline Makefile:57]: We do not only generate constants, but also generate the security scan config. Maybe call it POPULATE_IMAGES or something?

File: .golangci.yaml (hunk 3 above)
Comment (hisarbalik) [reply .golangci.yaml]: not required, removed from aliases

File: Makefile (hunk 4 above)
Comment (hisarbalik) [reply Makefile]: good point, I renamed it to `POPULATE_IMAGES`

File: internal/constant/images.go [hunk 5]
```diff
@@ -0,0 +1,11 @@
+// This file is generated by "make generate".
+// Don't edit, update .env file and run make target generate.
+
+package constant
```
[the remark this answers is not in the document]
Comment (hisarbalik) [reply internal/constant/images.go]: Package renamed

**Review by NHingerl — APPROVED** [review]

**Review by shorim — APPROVED** [review]

**Pull request merged**